edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/MetricsWorker.cs

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public class MetricsWorker : IDisposable
```

Comment:
Need to know about its responsbility for better review; please provide high level design/class relationships.

Reply:
I'm not sure what you mean by this. when you get a chance, tap my shoulder and well talk.

Reply:
Please update this PR description with the spec link.